resources/azure/environmentDefaults.go

```diff
@@ -43,8 +46,11 @@ func sandboxDefault() environmentDefault {
 			defaultVNet:            "/subscriptions/8c56d827-5f07-45ce-8f2b-6c5001db5c6f/resourceGroups/datadog-agent-testing/providers/Microsoft.Network/virtualNetworks/default-vnet",
 			defaultSubnet:          "/subscriptions/8c56d827-5f07-45ce-8f2b-6c5001db5c6f/resourceGroups/datadog-agent-testing/providers/Microsoft.Network/virtualNetworks/default-vnet/subnets/default-subnet",
 			defaultSecurityGroup:   "/subscriptions/8c56d827-5f07-45ce-8f2b-6c5001db5c6f/resourceGroups/datadog-agent-testing/providers/Microsoft.Network/networkSecurityGroups/default",
-			defaultInstanceType:    "Standard_B4ms",
-			defaultARMInstanceType: "Standard_D4ps_v5",
+			defaultInstanceType:    "Standard_D4s_v3",  // Allows nested virtualization for kata runtimes
+			defaultARMInstanceType: "Standard_D4ps_v5", // No azure arm instance supports nested virtualization
+			defaultKataOsSku:       "Mariner",
```

Comment:
Generally speaking, these options are AKS specific. They should in their own namespace (see `eks` on AWS): `az/aks/linuxKataNodeGroup`

The doc says it only supports `AzureLinux`: https://learn.microsoft.com/en-us/azure/aks/use-pod-sandboxing#deploy-to-an-existing-cluster. Also I'm not sure we need to input an option for that.

Reply:
I think Mariner also works. If I remember correctly I tried with the default setting and it said something about only `Mariner` being supported.

Reply:
I think both Mariner and AzureLinux are supported. I removed this option and hardcoded AzureLinux.